### DISCUSSION TITLE
CarPlay: speed selection

### DISCUSSION THREAD

**Discussion by SpikeLightfoot** [TortugaPower/BookPlayer#641] [Ideas]
I love this app and use it to practice Spanish on my daily commute.  Sometimes I need to slow the playback so I can understand the spoken Spanish better, but I can only do that from the interface on my phone; the CarPlay interface shows no speed controller.  I'd love to see that addition, as I don't like looking down at my phone while I'm driving.

**Comment by GianniCarlo** [comment]
Hi @SpikeLightfoot , this was considered for a time, but there's a limitation here on the speed options that CarPlay will let us show, right now in the UI (I don't remember the exact details), but it jumps 0.25 intervals from 0.5 to 2 (I think). So if the user is using another speed option, it starts to mess the UI and logic around. 

I know that for audio apps, there is currently a new way to do the library interface since iOS 13 (not necessarily the now playing screen though), but I haven't looked into that yet, maybe things have changed(?)

**Reply by SpikeLightfoot** [reply]
Thanks for the reply!
From the top: in the app 3.0, 2.5, 2.0, 1.75, 1.5, 1.25, 1.15, 1.1. 1.0, 0.9, 0.75. 0.5X.
For me that's way too much granularity.  I'd rather have something like 2, 1.5, 1, 0.75.  If that makes it easier programmatically I'm all in favor.  I don't know what you mean by "another speed option."  Is there another way to control playback speed?
